DataType/Base/DataType.php

```diff
@@ -30,6 +30,18 @@ public function __construct(\obo\Carriers\PropertyInformationCarrier $propertyIn
     public abstract function name();
 
     /**
+     * @param mixed $value
+     * @return void|\obo\Exceptions\Exception
+     */
+    public function validate($value) {
+        foreach ($this->propertyInformation->annotations as $annotation) {
```

Comment:
If use information from propertyInformation, you will not have to iterate annotations